### PR TITLE
When no fcp devices in zvm, smcli should return empty

### PR DESCRIPTION
**Pull request by IBMCaoBiao**
Signed-off-by: CaoBiao <bjcb@cn.ibm.com>

### CODE REVIEW

File: smtLayer/getVM.py [hunk 1]
```diff
@@ -663,7 +663,13 @@ def fcpinfo(rh):
         # write the ret into results['response']
         rh.printLn("N", ret)
     else:
-        rh.printLn("ES", results['response'])
-        rh.updateResults(results)    # Use results from invokeSMCLI
+        if 'Return buffer is empty' in ''.join(results['response']):
```
Comment (IBMCaoBiao) [inline smtLayer/getVM.py:666]: will change to use return and reason code

**Comment by IBMCaoBiao** [comment]
# The return code investigation

## 1. the actual return code for this issue:
```
[root@t4605001 ~]# sudo /opt/zthin/bin/smcli System_WWPN_Query --addRCheader -T IAAS0276 -k OWNER=YES
8 4 28 (details) ULGSMC5004W Return buffer is empty
  API issued : System_WWPN_Query
Failed
  Return Code: 4
  Reason Code: 28
  Description: ULGSMC5004W Return buffer is empty
  API issued : System_WWPN_Query
[root@t4605001 ~]#
```

## 2. the description in Redboot
Screenshot 2022-11-23 at 12 40 34

## 3. Conclusion: we can use the return code `4` and reason code `28` as the identity to locate the problem.

File: smtLayer/getVM.py [hunk 2]
```diff
@@ -663,7 +663,15 @@ def fcpinfo(rh):
...
     else:
-        rh.printLn("ES", results['response'])
-        rh.updateResults(results)    # Use results from invokeSMCLI
+        if results['rc'] == 4 and results['rs'] == 28:
+            # In this case, Return buffer is empty. Means there is no FCP
+            # devices in z/VM, so we will reset results and return empty
+            # content, so that the upper layer can not sense the errors.
+            rh.printSysLog("In changeVM.dedicate, not found fcp "
+                           "devices in z/VM, will return empty.")
+            rh.updateResults(results, reset=2)
```
Comment (dongyanyang) [inline smtLayer/getVM.py:672]: instead of using reset, why would we keep line 655 to line 658? can we just remove that code?

Comment (IBMCaoBiao) [reply smtLayer/getVM.py:672]: ok, I will remove the duplicated code.

**Comment by IBMCaoBiao** [comment]
# test results: https://github.ibm.com/zvc/planning/issues/10967#issuecomment-50749625

File: smtLayer/getVM.py [hunk 3]
```diff
@@ -652,18 +652,30 @@ def fcpinfo(rh):
...
+        if results['rc'] == 4 and results['rs'] == 28:
+            # In this case, Return buffer is empty. Means there is no FCP
+            # devices in z/VM, so we will reset results and return empty
+            # content, so that the upper layer can not sense the errors.
+            rh.printSysLog("In changeVM.dedicate, not found fcp "
+                           "devices in z/VM, will return empty.")
+            new_results = dict()
+            new_results['overallRC'] = 0
+            new_results['rc'] = 0
+            new_results['rs'] = 0
```
Comment (dongyanyang) [inline smtLayer/getVM.py:670]: keep the original rc and rs, instead of using 0

**Review by dongyanyang — APPROVED** [review]
LGTM

**Pull request merged**